sequencer/src/main/java/net/consensys/linea/sequencer/txpoolvalidation/LineaTransactionPoolValidatorPlugin.java

```diff
@@ -113,6 +121,45 @@ public void start() {
               l1L2BridgeSharedConfiguration(),
               rejectedTxJsonRpcManager));
 
+      if (metricCategoryRegistry.isMetricCategoryEnabled(TX_POOL_PROFITABILITY)) {
```

Comment:
Similarly to how it's done for `ProfitableTransactionSelector` maybe instead of evaluating each transaction from txpool directly, maybe it would be more efficient and less invasive if we capture txpool profitability metrics in `ProfitabilityValidator`?

Reply:
we thought about that, but since a tx could stay in the pool for a long time, the value calculated at `ProfitabilityValidator` time could be different from the current value, due to the changes to the market conditions and the extra data pricing conf, so basically it will be a different metric, that represents the values at validation time, instead of the current status in the txpool. 
If you think that collecting metrics in the `ProfitabilityValidator` is good enough, then yes it will be more efficient.

Reply:
I see what you mean. Let's leave it the way you did and if it causes any performance issues, we can always disable it